atoma-auth/src/auth.rs

```diff
@@ -571,18 +688,30 @@ impl Auth {
             .get_wallet_address()
             .map_err(AuthError::SuiError)?;
         if receiver == own_address {
-            let (result_sender, result_receiver) = oneshot::channel();
-            self.state_manager_sender
-                .send(AtomaAtomaStateManagerEvent::ConfirmUser {
-                    sui_address: sender.to_string(),
-                    user_id: claims.user_id,
-                    result_sender,
-                })?;
-            let is_their_wallet = result_receiver.await??;
-
-            if !is_their_wallet {
-                return Err(AuthError::PaymentNotForThisUser);
+            match proof_signature {
```

Comment:
Can you explain where does how is this `proof_signature` is generated ? And in which scenarios is it present in the current method body scope and when it is not ?